patches/0005-Adjust-Go-tests-to-work-with-crypto-module.patch

```diff
@@ -0,0 +1,95 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: microsoft-golang-bot <microsoft-golang-bot@users.noreply.github.com>
+Date: Tue, 15 Mar 2022 18:30:30 -0500
+Subject: [PATCH] Adjust Go tests to work with crypto module
+
+---
+ src/cmd/go/testdata/script/gopath_std_vendor.txt |  9 ++++++++-
+ src/cmd/link/internal/ld/lib.go                  |  2 +-
+ src/crypto/boring/boring_test.go                 |  1 +
+ src/go/build/deps_test.go                        | 13 ++++++++++---
+ 4 files changed, 20 insertions(+), 5 deletions(-)
+
+diff --git a/src/cmd/go/testdata/script/gopath_std_vendor.txt b/src/cmd/go/testdata/script/gopath_std_vendor.txt
+index a0a41a50de..2d28bd2cf9 100644
+--- a/src/cmd/go/testdata/script/gopath_std_vendor.txt
++++ b/src/cmd/go/testdata/script/gopath_std_vendor.txt
+@@ -23,7 +23,14 @@ go list -deps -f '{{.ImportPath}} {{.Dir}}' .
+ stdout $GOPATH[/\\]src[/\\]vendor[/\\]golang.org[/\\]x[/\\]net[/\\]http2[/\\]hpack
+ ! stdout $GOROOT[/\\]src[/\\]vendor
+ 
+-go list -test -deps -f '{{.ImportPath}} {{.Dir}}' .
++# Use build tag 'gocrypt' while evaluating test dependencies to avoid importing
++# vendored crypto module dependencies like go-crypto-openssl. This test script
++# is not set up to handle any vendored libraries being imported other than
++# golang.org/x/net/http2/hpack, so we must make sure it is the only one.
++#
++# See https://github.com/microsoft/go/issues/481 for more details, such as the
++# dependency chain that would cause the failure if the gocrypt tag isn't used.
++go list -tags=gocrypt -test -deps -f '{{.ImportPath}} {{.Dir}}' .
```

Comment:
Note: I had to add `-tags=gocrypt` to `go list`, not `go build`, because this list isn't actually based on the binary. On the plus side: this is a more minimal change. 😄